psi4/src/psi4/libmints/wavefunction.cc

```diff
@@ -784,18 +792,18 @@ void Wavefunction::set_reference_wavefunction(const std::shared_ptr<Wavefunction
 }
 
 void Wavefunction::force_doccpi(const Dimension &doccpi) {
+    auto socc = soccpi();
     for (int h = 0; h < nirrep_; h++) {
-        if ((soccpi_[h] + doccpi[h]) > nmopi_[h]) {
+        if ((socc[h] + doccpi[h]) > nmopi_[h]) {
             throw PSIEXCEPTION(
                 "Wavefunction::force_doccpi: Number of doubly and singly occupied orbitals in an irrep cannot exceed "
                 "the total number of molecular orbitals.");
         }
-        doccpi_[h] = doccpi[h];
-        nalphapi_[h] = doccpi_[h] + soccpi_[h];
-        nbetapi_[h] = doccpi_[h];
+        nalphapi_[h] = doccpi[h] + socc[h];
+        nbetapi_[h] = doccpi[h];
     }
-    nalpha_ = doccpi_.sum() + soccpi_.sum();
-    nbeta_ = doccpi_.sum();
+    nalpha_ = nalphapi_.sum();
+    nbeta_ = nbetapi_.sum();
```

Comment:
Shouldn't there be a spin state check here?

Reply:
I don't know. The `force` functions are [marked](https://github.com/psi4/psi4/blob/master/psi4/src/psi4/libmints/wavefunction.h#L402-L416) as intended for expert use only because careless use can produce inconsistencies. Nothing in Psi uses them. I can't tell whether the potential to cause a spin state violation is a bug or a feature. Maybe a warning is in order if the number of unpaired electrons changes?

Reply:
OK, in that case we can forget about it

Reply:
Actually... I think the `force` functions may break with these changes, since `socc` and `docc` are recomputed. One would have to override `socc` and `docc` simultaneously lest the occupations be changed in the other code.

Reply:
I would remove the force functions altogether, since it sounds like they aren't even used..

Reply:
@loriab Fine with you? These functions are dangerous. Forte doesn't use them. V2RDM doesn't use them? Psi doesn't use them.

Reply:
I need to look further. I think there was a Shannon Houck change that allowed resetting docc/socc for detci, and CDS agreed it was a legit use. Would have to see if this is that.

Reply:
aha, https://github.com/psi4/psi4/pull/876 . the ability added in that PR should be preserved. it likely enables things in the fockCI plugin

Reply:
I'll give that a look. Do we have FockCI test coverage?

Reply:
* The express purpose of this forcing is to [change spin state](https://github.com/shannonhouck/psi4fockci/blob/267e17103c77dcd42cf1a66a5b805db3bcd5ed9e/psi4fockci/spinflip.py#L131-L142). So the lack of a spin state check is definitely a feature and not a bug.
* This use cases forces both `doccpi` and `soccpi`. It makes sense to me to have the user force both simultaneously.

Reply:
Then those functions definitely need to be merged into `force_docc_socc`; otherwise there is no guarantee that the code will actually use the input arrays.

Reply:
> I'll give that a look. Do we have FockCI test coverage?

possibly not, 😳 . She was working on two plugins, old and new. It was near a psi release, and I got very confused and packaged the wrong one. She added https://github.com/psi4/psi4/blob/master/doc/sphinxman/source/plugin_psi4fockci.rst though, so the internal build and that input might work.

Reply:
a combo docc_and_socc can work: https://github.com/shannonhouck/psi4fockci/commit/9a3916598030cdd501c7370ca2040bd7659e018b#diff-e33326fe2aaf9f7d7d385252ae30fa0c5a6d47751f890592c8ddc82bd8b861e4R83

Reply:
> > I'll give that a look. Do we have FockCI test coverage?
> 
> possibly not, 😳 . She was working on two plugins, old and new. It was near a psi release, and I got very confused and packaged the wrong one. She added https://github.com/psi4/psi4/blob/master/doc/sphinxman/source/plugin_psi4fockci.rst though, so the internal build and that input might work.

What does "internal build" mean?

Reply:
toggle `ENABLE_psi4fockci=ON` in `<objdir>/CMakeCache.txt` and rebuild. hopefully cmake will pull a copy and install it into `stage/` so it's ready to run.

Reply:
@loriab It doesn't look like FockCI has any tests registered with Psi. Am I mistaken?

Reply:
you could try https://github.com/psi4/psi4/blob/master/tests/pytests/test_addons.py#L1236 . because of my repo confusion, though, it may not be what you're after.